java/FoodTruckApp2/src/main/java/com/my/FoodTruckApp2/Order/OrdersRepository.java

```diff
@@ -0,0 +1,178 @@
+    // appetizers & entrees(NOT THERE IDs) that the customer ordered).
+    // For Example:
+    //OrderDto orderDto = new OrderDto(order.getId(), ordersRequestBody.getCustomerId(),
+    // appetizerOrders, entreeOrders);
+//
+    public OrderDto CreateNewOrder(NewOrderRequestBody ordersRequestBody) {
+        // first when the order is created (Inserted into order) we need to find the appetizers
+        String sql = "INSERT INTO \"order\" (customer_id) VALUES (?) RETURNING *";
+        Orders order = jdbcTemplate.queryForObject(sql, new BeanPropertyRowMapper<>(Orders.class),
+                ordersRequestBody.getCustomerId());
+        log.info("Order is Created");
+        // the order is inserted now we take the appetizerId from the requestBody
+        // now we need to find the id by using the method in AppetizerRepository
+        log.info("Looking for the appetizers...");
+        List<Appetizer> appetizerOrders = new ArrayList<>();
+        List<Integer> listOfAppetizerIds = ordersRequestBody.getAppetizerIds();
+        System.out.println("List of appetizersIds: " + listOfAppetizerIds);
+        listOfAppetizerIds.forEach(appetizerId -> {
+            Appetizer appetizers =
+                    appetizerRepository.gettingAppetizerById(appetizerId);
+            System.out.println("Found appetizer " + appetizers);
+            String sqlAppetizer = " INSERT INTO appetizer_ordered(order_id ,appetizer_id) VALUES (?,?) RETURNING *";
+            AppetizerOrdered appetizerOrdered = jdbcTemplate.queryForObject(sqlAppetizer,
+                    new BeanPropertyRowMapper<>(AppetizerOrdered.class),
+                    order.getId(), appetizers.getId());
+            log.info("Appetizer receipt: " + appetizerOrdered);
+            appetizerOrders.add(appetizers);
+            System.out.println("The list of appetizers the customer ordered: " + appetizerOrders);
+        });
+        log.info("Looking for the entrees...");
+        List<Entree> entreeOrders = new ArrayList<>();
+        List<Integer> listOfEntreeIds = ordersRequestBody.getEntreeIds();
+        listOfEntreeIds.forEach(entreeId -> {
+            Entree entree =
+                    entreeRepository.gettingEntreeById((entreeId));
+            System.out.println("Found Entree " + entree);
+            String sqlEntree = " INSERT INTO entree_ordered(order_id ,entree_id) VALUES (?,?) RETURNING *";
+            EntreeOrdered entreeOrdered = jdbcTemplate.queryForObject(sqlEntree,
+                    new BeanPropertyRowMapper<>(EntreeOrdered.class),
+                    order.getId(), entree.getId());
+            log.info("Entree receipt: " + entreeOrdered);
+            entreeOrders.add(entree);
+            System.out.println("The list of entrees the customer ordered: " + entreeOrders);
+        });
+        OrderDto orderDto = new OrderDto(order.getId(), ordersRequestBody.getCustomerId(), appetizerOrders, entreeOrders);
+        return orderDto;
+    }
```

Comment:
1. You have a lot of sysouts here. Either delete them or make them logs
2. This method name should not be uppercased.
3. Biggest problem here is you have a lot of logic in this repo call that don't belong here. This method should be creating an order, and nothing else. The rest of this belongs in the service instead

Reply:
Ok

Reply:
it looks like 3 is still an issue

Reply:
Really! i thought you were talking about the inserts, maybe i missed something . Can you please point out the issue so i can resolve it